contributors/design-proposals/storage/dynamic_volume_limit.md

```diff
@@ -0,0 +1,67 @@
+# Dynamic attached volume limits
+
+## Goals
+
+Currently the number of volumes attachable to a node is either hardcoded or only configurable via an environment variable. Also
+existing limits only apply to well known volume types like EBS, GCE and is not available to all volume plugins.
+
+This proposal enables any volume plugin to specify those limits and also allows same volume type to have different volume
+limits depending on type of node.
+
+## Implementation Design
+
+### Prerequisite
+
+* This feature will be protected by an alpha feature gate, so as API and CLI changes needed for it. We are planning to call
+  the feature `DynamicVolumeThreshold`.
+
+### API Changes
+
+We propose following changes to node's status field:
+
+```
+node.Status.AttachedVolumeLimits map[string]int{}
+node.Status.AttachedVolumeLimit: {
+  "kubernetes.io/aws-ebs": 39
+}
+```
+
+For CSI the name used in key will be - `kubernetes.io/csi/<driver_name>`
+
+### Changes to CSI
```

Comment:
Remove CSI call outs for now.

Reply:
The design should move the 'default values' from the volume plugin to an admin modifiable location (such as CLI args).   The admin should set them per node as part of configuration.

We should not conflate CSI spec with this design.  Once we have the design correct for kubernetes, then we look at changes to CSI.